src/index.ts

```diff
@@ -0,0 +1,106 @@
+  }
+}
+
+/**
+ * The possible options to configure the Checkout SDK.
+ */
+interface CheckoutOptions {
+  /** The id of the HTML element in which to insert the Checkout form. */
+  id?: string
+  /** The callback to call on Checkout events. */
+  onMessage?: (message: string) => void
+  /** Whether the SDK should use testing APIs. */
+  testing?: boolean
+}
+
+/** The default values for all the options. */
+const defaultOptions: CheckoutOptions = {
+  id: 'easypay-checkout',
+  onMessage: () => { /* do nothing */ },
+  testing: false,
+}
+
+const LOGTAG = '[easypay Checkout SDK]'
+
+/** Validates the necessary parameters for Checkout initialization and gives helpful messages for integrators. */
+function validateParameters(manifest: CheckoutManifest, options: CheckoutOptions) {
+  if (!manifest || typeof manifest.id !== 'string' || manifest.id === '' || typeof manifest.session !== 'string' || manifest.session === '') {
+    console.error(`${LOGTAG} Please provide a valid Checkout Manifest when calling startCheckout.`)
+    return false
+  }
+  if (typeof options.id !== 'string' || options.id === '') {
+    console.error(`${LOGTAG} The HTML element id must be a non-empty string.`)
+    return false
+  }
+  if (typeof options.onMessage !== 'function') {
+    console.error(`${LOGTAG} The onMessage callback must be a function.`)
+    return false
+  }
+  return true
+}
+
+/** Encodes a Manifest for iframe consumption. Supports both Node and browser environments. */
+function encodeManifest(manifest: CheckoutManifest) {
+  if (typeof Buffer !== 'undefined') {
+    return Buffer.from(JSON.stringify(manifest), 'binary').toString('base64')
+  } else {
+    return btoa(JSON.stringify(manifest))
+  }
+}
+
+/** Used to configure and populate the Checkout form. */
+export function startCheckout(manifest: CheckoutManifest, options: CheckoutOptions = {}) {
+  const mergedOptions = {...defaultOptions, ...options}
+  if (!validateParameters(manifest, mergedOptions)) {
+    return
+  }
+  let url = 'https://checkout-serverless.quality-utility.aws.easypay.pt'
+  if (mergedOptions.testing) {
+    url = 'https://checkout-serverless.sandbox.easypay.pt'
+  }
+  const listener = (e: MessageEvent) => {
+    if (e.origin === url && e.data.type === 'ep-checkout') {
+      options.onMessage!(e.data.status)
+      if (e.data.status === 'complete') {
+        window.removeEventListener('message', listener)
+      }
+    }
+  }
+
+  window.addEventListener('message', listener)
+  const iframe = document.createElement('iframe')
+  iframe.setAttribute('src', `${url}?manifest=${encodeManifest(manifest)}`)
+  // Using the attributes below in order to provide defaults without overriding CSS styles
+  iframe.setAttribute('width', '400')
+  iframe.setAttribute('height', '700')
+  iframe.setAttribute('frameborder', '0')
+  const el = document.getElementById(mergedOptions.id!)
+  if (!el) {
+    console.error(`${LOGTAG} Could not find element ${mergedOptions.id}.`)
```

Comment:
This is what I'm saying to do on top. :)

Reply:
Ok, I can move it.